scripts/runDiscoPoP

```diff
@@ -0,0 +1,314 @@
+    *)
+      echo -e "${CYA}DP_DBUG:${DEF} $*"
+      ;;
+  esac
+}
+
+
+#####
+# show configuration
+#####
+
+log -i \
+  "CONFIGURATION:
+  llvm-clang  $LLVM_CLANG
+      $($LLVM_CLANG --version | grep version)
+  llvm-clang++ $LLVM_CLANGPP
+      $($LLVM_CLANGPP --version | grep version)
+  llvm-ar     $LLVM_AR
+      $($LLVM_AR --version | grep version)
+  llvm-link   $LLVM_LINK
+      $($LLVM_LINK --version | grep version)
+  llvm-dis    $LLVM_DIS
+      $($LLVM_DIS --version | grep version)
+  llvm-opt    $LLVM_OPT
+      $($LLVM_OPT --version | grep version)
+  llvm-llc    $LLVM_LLC
+      $($LLVM_LLC --version | grep version)
+  gllvm:      $GLLVM
+  project:    $PROJECT
+  ldFlags:    $LINKER_FLAGS
+  execName:   $EXECUTABLE_NAME
+  execArgs:   $EXECUTABLE_ARGUMENTS
+  threads:    $THREADS
+  discoPoP:   $DP
+  dp_build:   $DP_BUILD
+  gllvmLog:   $GLLVM_LOG"
+
+#####
+# configure gllvm
+#####
+
+#export LLVM_COMPILER_PATH="<path/to/llvm/build>"
+export LLVM_CC_NAME=$LLVM_CLANG
+export LLVM_CXX_NAME=$LLVM_CLANGPP
+export LLVM_AR_NAME=$LLVM_AR
+export LLVM_LINK_NAME=$LLVM_LINK
+export WLLVM_OUTPUT_LEVEL="AUDIT"  # can be used to more easily determine flags needed during the link step
+export WLLVM_OUTPUT_FILE=$GLLVM_LOG
+export LLVM_BITCODE_GENERATION_FLAGS="-g -O0 -fno-discard-value-names"
+#export LLVM_LINK_FLAGS=""
+
+log -d "GLLVM SANITY CHECK:
+$($GLLVM/gsanity-check)" 
+
+#####
+# create FileMapping.txt,
+# build with gllvm,
+# create single .bc and .ll from the executable
+#####
+
+cd $PROJECT
+
+# create FileMapping.txt
+rm -f FileMapping.txt
+$DP/scripts/dp-fmap
+
+# build using gllvm as compiler
+log -i "Building your application..."
+log -d $(make CC=$GLLVM/gclang CXX=$GLLVM/gclang++ LD=$GLLVM/gclang++ -j$THREADS $MAKFILE_TARGET)
+
+# create single .bc and .ll from the executable
+log -d $($GLLVM/get-bc -b -m -v $EXECUTABLE_NAME)
+$LLVM_DIS ${EXECUTABLE_NAME}.bc -o ${EXECUTABLE_NAME}.ll
+
+#####
+# apply instrumentation and run
+#####
+
+# Run DiscoPoP Pass
+log -i "Running DiscoPoP Pass to analyze and instrument your application..."
+$LLVM_OPT -S -load ${DP_BUILD}/libi/LLVMDiscoPoP.so -DiscoPoP ${EXECUTABLE_NAME}.ll -o ${EXECUTABLE_NAME}_dp.ll -fm-path FileMapping.txt
+
+log -i "Creating Executable..."
+$LLVM_LLC -filetype=obj  ${EXECUTABLE_NAME}_dp.ll -o  ${EXECUTABLE_NAME}_dp.o
+$LLVM_CLANGPP ${EXECUTABLE_NAME}_dp.o -Wl,--export-dynamic -O0 -g -o ${EXECUTABLE_NAME}_dp -L$DP_BUILD/rtlib -lDiscoPoP_RT -lpthread $LINKER_FLAGS
+
+# another way to get the executable from the .ll file without going via the .o object file:
+#$LLVM_CLANGPP ${EXECUTABLE_NAME}_dp.ll -Wl,--export-dynamic -O0 -g -o ${EXECUTABLE_NAME}_dp -L$DP_BUILD/rtlib -lDiscoPoP_RT -lpthread $LINKER_FLAGS
+
+# Run DPReduction Pass
+log -i "Running DiscoPoP Reduction Analysis Pass to detect reduction operations..."
+$LLVM_OPT -S -load ${DP_BUILD}/libi/LLVMDPReduction.so -DPReduction ${EXECUTABLE_NAME}.ll -disable-output -fm-path FileMapping.txt
+
```

Comment:
```suggestion
```

Reply:
maybe remove it now to prevent it from getting forgotten once this branch and the removal of the DPReduction pass are merged.

Reply:
#147 has been closed.